src/cljs/org/broadinstitute/firecloud_ui/page/workspaces.cljs

```diff
@@ -129,19 +129,38 @@
   {:render
    (fn [{:keys [state refs]}]
      [:div {}
+      ;; Begin overlay
+      [:div {:style {:visibility (if-not (:overlay-shown? @state) "hidden")
+                     :position "absolute"
+                     :top "0px" :left "0px" :width "100%" :height "100%"
+                     :background "rgba(82, 129, 197, 0.4)" :zIndex "99999"
+                     :text-align "center"}}
+       [:div {:style {:width "300px" :margin "100px auto"
+                      :background-color "white" :border "1px solid black"
+                      :padding "15px"
+                      :text-align "center"}}
+        [:div {} "Enter a name for the new Workspace:"]
+        [:div {:style {:padding "10px"}} [:input {:ref "wsName" :type "text"}]]
+        (let [clear-overlay (fn []
+                              (swap! state assoc :overlay-shown? false)
+                              (set! (.-value (.getDOMNode (@refs "wsName"))) ""))]
```

Comment:
If you decide to go with `:display "none"` above, you'll need to move this statement up one line so you're not setting the value of a DOM node that is no longer present in the document.